Modules/ManageRDSessions/ManageRDSessions.psm1

```diff
@@ -117,11 +119,17 @@ function Remove-sbRDSession {
                     'Force'            = $true
                 }
 
-                Write-Verbose "Attempting Logoff of $($session.Username) on $($session.HostServer)"
-                Invoke-RDUserLogoff @params
-                Write-Host "User [$($session.Username)] logged off from [$($session.HostServer)]"
+                if ($AsJob.IsPresent) {
+                    Write-Verbose "Attempting Logoff of [$($session.Username)] on [$($session.HostServer)] [AsJob]"
+                    # Need to use "using:" scope here to pass local hashtable to Job function, otherwise will pass all as null
+                    $sb = {Invoke-RDUserLogoff @using:params}
+                    Start-Job -ScriptBlock $sb -Name "Log Off [$($session.UserName)]" | Select-Object Id,Name,State
```

Comment:
If you're returning object, don't use Select or Format-* commands; just the object itself.
Remove ` | Select-Object Id,Name,State`

Reply:
Understood - and thanks for the further explanation :)